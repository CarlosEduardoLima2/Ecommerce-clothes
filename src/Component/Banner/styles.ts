import styled, { css } from 'styled-components';
import {Container as TextComponent} from '../TextComponent/styles'
import {Container as Heading} from '../Heading/styles'


export const Container = styled.section`
  ${({theme}) => css`
    width: 100%;
    height: 100vh;
    background-color: ${theme.colors.primaryColor};

    @media (max-width: 880px){
    height: 100%;
    }
  `}
`;

export const Content = styled.div`
  ${({theme}) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 16px;
    gap: 2rem;

    .swiper-slide {
      display: flex;
      justify-content: center;
      background: #fff;
      background: transparent;
      align-items: center;


      div{
        margin: 30px 0;
      }

      ${TextComponent}{
        display: block;
        max-width: 500px;
        margin: 15px 0;
        font-size: 0.8rem;
      }

    }

    @media (max-width: 460px) {
      flex-wrap: wrap;
    }
    
  `}
`;

export const Text = styled.section`
  
`;
export const Image = styled.div`
  ${({theme}) => css`
    max-width: 350px;
    max-height: 600px;
    img{
      object-fit: cover;
      object-position: center;
      height: 100%;
    }

    @media (max-width: 880px){
      img{
        width: 80%;
        height: 80%;
      }
    }
  `}
`;