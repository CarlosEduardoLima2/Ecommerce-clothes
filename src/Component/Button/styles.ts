
import styled, { css } from 'styled-components';
import { NavLink } from 'react-router-dom';

export const Container = styled(NavLink)`
  ${({theme}) => css`
    padding: 10px 60px;
    border-radius: 8px;
    color: ${theme.colors.lightColor};
    background-color: ${theme.colors.darkColor};
    text-decoration: none;
    transition: .3s ease-in;

    &:hover{
      border: none;
      background: #363636;
    }
  `}
    
`;
