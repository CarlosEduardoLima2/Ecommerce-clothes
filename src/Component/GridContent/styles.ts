import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit,minMax(250px,1fr));
  gap: 88px 30px;
`;
