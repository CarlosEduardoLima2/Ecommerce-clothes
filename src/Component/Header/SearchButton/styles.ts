import styled from 'styled-components';

export const Container = styled.button`
  border: none;
  background-color: #fff;
  transition: .4s ease;

  &:hover{
    opacity: 0.6;
  }
`;
