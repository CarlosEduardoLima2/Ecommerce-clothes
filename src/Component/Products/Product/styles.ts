import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';

export const Container = styled(Link)`
  height: 378px;
  text-decoration: none;
`;

export const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
`

export const ImageContainer = styled.div`
  border-radius: 35px;
  background-color: #dadad8;
  height: 369px;
  text-align: center;
`
export const Content = styled.div`
  max-width: 100%;
  margin-top: 17px;
  display: flex;
  justify-content: space-between;
`

export const ContentText = styled.div`

`;

export const Price = styled.span`
  ${({theme}) => css`
    text-decoration: none;
    color: ${theme.colors.darkColor};
    font-size: ${theme.fonts.sizes.small};
    font-weight: 400;
    margin-left: 10px;
  `}
`;

export const NewProduct = styled.span`
  ${({theme}) => css`
    color: ${theme.colors.secondaryColor};
    font-weight: 400;
    margin-bottom: 4px;
    font-size: ${theme.fonts.sizes.xxsmall};
  `}
`
export const Title = styled.span`
  ${({theme}) => css`
    font-weight: 700;
    text-decoration: none;
    color: ${theme.colors.darkColor};
    font-size: ${theme.fonts.sizes.xsmall};
  `}
`