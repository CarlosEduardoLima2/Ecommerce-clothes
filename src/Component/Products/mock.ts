export const mock = {
    "id": 0,
    "title": "Camisa com logo",
    "description": "Se todo o seu guarda-roupa é preto e pelo menos uma pessoa em sua vida lhe disse para usar mais cores, esta camisa é o compromisso que você estava esperando.",
    "price": 40,
    "category": "vestido",
    "sizes": [
      "XS",
      "S",
      "M",
      "L",
      "XL",
      "2XL"
    ],
    "image": "./assets/pexels-abir-hasan-1857375-removebg-preview.png",
    "newProduct": true
  };
