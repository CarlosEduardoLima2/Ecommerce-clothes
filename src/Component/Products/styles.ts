import styled from 'styled-components';

export const Container = styled.div`
  margin: 8rem 0;
`;
