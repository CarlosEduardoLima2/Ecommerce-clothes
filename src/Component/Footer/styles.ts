
import styled from 'styled-components';
import {Container as Text} from "../TextComponent/styles"
import {Container as Link} from "../NavLinkComponent/styles"
import { theme } from '../../styles/theme';

export const Container = styled.div`
  background-color: ${theme.colors.primaryColor};
  width: 100%;
`;

export const Info = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;

  @media ${theme.media.lteMedium} {
    justify-content: center;
  }
`;

export const DataContainer = styled.div`
  padding: 1.5rem 0;

  ${Text}{
    margin: 1rem 0;
  }
`;

export const Icons = styled.div`

`;
export const InfoText = styled.div`

`;

export const LinksContainer = styled.div`
  display: flex;
  flex-direction: column;

  ${Link}{
    padding: 0.5rem;
  }
`;

export const About = styled.div`
  max-width: 300px;
  padding: 1rem;
  text-align: center;
  ${Text}{
    margin: 1rem 0;
  }
`;
