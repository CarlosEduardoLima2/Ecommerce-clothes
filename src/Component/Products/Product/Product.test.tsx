import { screen } from '@testing-library/react';
import { Product } from '.';
import { renderTheme } from '../../../styles/render-theme';
import { mock } from '../mock';

describe('<Product />', () => {
  it('should mach snapshot', () => {
    const { container } = renderTheme(<Product to="/" {...mock}/>);
    expect(screen.getByText('Camisa com logo')).toBeInTheDocument();
    expect(screen.getByRole('img', {name: "Camisa com logo"})).toBeInTheDocument();
    expect(screen.getByText('New!')).toBeInTheDocument();
    expect(screen.getByText('$40')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
