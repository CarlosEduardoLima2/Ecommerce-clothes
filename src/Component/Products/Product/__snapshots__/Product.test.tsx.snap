// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> should mach snapshot 1`] = `
.c0 {
  height: 378px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  max-width: 100%;
  max-height: 100%;
}

.c1 {
  border-radius: 35px;
  background-color: #dadad8;
  height: 369px;
  text-align: center;
}

.c3 {
  max-width: 100%;
  margin-top: 17px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #212121;
  font-size: 0.875rem;
  font-weight: 400;
  margin-left: 10px;
}

.c4 {
  color: #10a958;
  font-weight: 400;
  margin-bottom: 4px;
  font-size: 1.25rem;
}

.c5 {
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #212121;
  font-size: 1rem;
}

<div>
  <a
    class="c0"
    href="/"
  >
    <div
      class="c1"
    >
      <img
        alt="Camisa com logo"
        class="c2"
        src="./assets/pexels-abir-hasan-1857375-removebg-preview.png"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class=""
      >
        <span
          class="c4"
        >
          New!
        </span>
        <div>
          <span
            class="c5"
          >
            Camisa com logo
          </span>
        </div>
      </div>
      <span
        class="c6"
      >
        $
        40
      </span>
    </div>
  </a>
</div>
`;
